internal/datasource/config_writer.go

```diff
@@ -0,0 +1,90 @@
+package datasource
+
+import (
+	"fmt"
+	"log/slog"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/nginx/agent/v3/api/grpc/instances"
+	"github.com/nginx/agent/v3/internal/client"
+	"github.com/nginx/agent/v3/internal/datasource/os"
+)
+
+type (
+	ConfigWriterParameters struct {
+		configDownloader client.HttpConfigClientInterface
+	}
+
+	ConfigWriter struct {
+		configDownloader client.HttpConfigClientInterface
+	}
+)
+
+func NewConfigWriter(configWriterParameters *ConfigWriterParameters) *ConfigWriter {
+	if configWriterParameters == nil {
+		configWriterParameters.configDownloader = client.NewHttpConfigClient()
+	}
+
+	return &ConfigWriter{
+		configDownloader: configWriterParameters.configDownloader,
+	}
+}
+
+func (cw *ConfigWriter) Write(lastConfigApply map[string]*instances.File, filesUrl string, tenantID uuid.UUID) (currentConfigApply map[string]*instances.File, skippedFiles map[string]struct{}, err error) {
+	currentConfigApply = make(map[string]*instances.File)
+	skippedFiles = make(map[string]struct{})
+
+	filesMetaData, err := cw.configDownloader.GetFilesMetadata(filesUrl, tenantID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error getting files metadata, filesUrl: %v, error: %v", filesUrl, err)
+	}
+
+filesLoop:
```

Comment:
Are we sure that `filesLoop` is necessary, as there is only one loop? My understanding was that named loops are used to break or continue an outer loop when in an inner loop 🤔

Reply:
Think this was left in by accident, Thank you!